### PR TITLE
Rtcninja

### PR DESCRIPTION
**Pull request by PetroccoCo**

### CODE REVIEW

File: modules/keyboardshortcut/keyboardshortcut.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ var KeyboardShortcut = {
                 $(":focus").is("input[type=password]") ||
                 $(":focus").is("textarea"))) {
                 if(e.which === "T".charCodeAt(0)) {
-                    if(APP.RTC.localAudio.isMuted()) {
+                    if(APP.RTC && APP.RTC.localVideo && APP.RTC.localAudio.isMuted()) {
```
Comment (watsoncj) [inline modules/keyboardshortcut/keyboardshortcut.js]: Is this a typo? (localVideo/localAudio)

Comment (PetroccoCo) [inline modules/keyboardshortcut/keyboardshortcut.js]: bad copy pasta job.

Comment (watsoncj) [inline modules/keyboardshortcut/keyboardshortcut.js]: :spaghetti:

File: modules/RTC/RTCUtils.js [hunk 2]
```diff
@@ -210,11 +210,44 @@ function RTCUtils(RTCService)
                 return this.audioTracks;
             };
         }
-    }
-    else
-    {
+    } else if (window.rtcninja) {
+      console.log('This appears to be an rtcninja platform');
```
Comment (watsoncj) [inline modules/RTC/RTCUtils.js]: This `console.log` should be wrapped in a `try/catch` to maintain existing browser support. (see line 250)

File: modules/UI/UI.js [hunk 3]
```diff
@@ -745,14 +745,19 @@ UI.setInitialMuteFromFocus = function (muteAudio, muteVideo) {
  * Mutes/unmutes the local video.
  */
 UI.toggleVideo = function () {
-    setVideoMute(!APP.RTC.localVideo.isMuted());
+    var newMuteState = !(APP.RTC && APP.RTC.localVideo && APP.RTC.localVideo.isMuted());
+    Analytics.sp.track('Video Mute Toggle', {
```
Comment (watsoncj) [inline modules/UI/UI.js]: `Analytics` is undefined. Arguably, we should be tracking this at the UI layer. @dansomething, has some cool angular directives that could be adapted to whatever analytics lib is pulled in at the app level.

File: modules/xmpp/TraceablePeerConnection.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
+var RTCUtils = require("../RTC/RTCUtils.js");
 function TraceablePeerConnection(ice_config, constraints) {
     var self = this;
-    var RTCPeerconnection = navigator.mozGetUserMedia ? mozRTCPeerConnection : webkitRTCPeerConnection;
-    this.peerconnection = new RTCPeerconnection(ice_config, constraints);
+    var rtcUtils = new RTCUtils(this);
+    this.peerconnection = new rtcUtils.peerconnection(ice_config, constraints);
```
Comment (watsoncj) [inline modules/xmpp/TraceablePeerConnection.js:5]: nice simplification!

File: modules/UI/UI.js (hunk 3 above)
Comment (dwilson6) [inline modules/UI/UI.js]: This track call should be removed

**Comment by PetroccoCo** [comment]
@watsoncj I think I've fixed up all the issues, wanna take a peak when you get a chance.

**Comment by watsoncj** [comment]
It sounds like we're waiting to get the rtcninja stream support before merging.

**Pull request closed without merging**